pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -92,12 +93,17 @@ type Config struct {
 	// Optional config for ALP
 	ALPEnabled bool
 
+	// Optional config for SidecarInjection
+	SidecarInjectionEnabled bool
+
 	// Calculated internal fields.
-	proxyImage      string
-	collectorImage  string
-	dikastesImage   string
-	dikastesEnabled bool
-	envoyConfigMap  *corev1.ConfigMap
+	proxyImage            string
+	collectorImage        string
+	dikastesImage         string
+	dikastesEnabled       bool
+	envoyEnabled          bool
```

Comment:
I assume `envoyEnabled` means `perHostEnvoyEnabled`? 

I think we should refactor these names to be explicit about whether they are referring to per-host, sidecar, or "any" version of the component for clarity.